diode-server/netbox/virtualization.go

```diff
@@ -0,0 +1,190 @@
+	Name        string  `json:"name,omitempty"`
+	Slug        string  `json:"slug,omitempty"`
+	Description *string `json:"description,omitempty"`
+	Tags        []*Tag  `json:"tags,omitempty"`
+}
+
+// VirtualizationClusterType represents a Virtualization Cluster Type
+type VirtualizationClusterType struct {
+	ID          int     `json:"id,omitempty"`
+	Name        string  `json:"name,omitempty"`
+	Slug        string  `json:"slug,omitempty"`
+	Description *string `json:"description,omitempty"`
+	Tags        []*Tag  `json:"tags,omitempty"`
+}
+
+// VirtualizationCluster represents a Virtualization Cluster
+type VirtualizationCluster struct {
+	ID          int                         `json:"id,omitempty"`
+	Name        string                      `json:"name,omitempty"`
+	Type        *VirtualizationClusterType  `json:"type,omitempty" mapstructure:"type"`
+	Group       *VirtualizationClusterGroup `json:"group,omitempty" mapstructure:"group"`
+	Site        *DcimSite                   `json:"site,omitempty"`
+	Status      *string                     `json:"status,omitempty"`
+	Description *string                     `json:"description,omitempty"`
+	Tags        []*Tag                      `json:"tags,omitempty"`
+}
+
+// Validate checks if the Virtualization Cluster is valid
+func (cluster *VirtualizationCluster) Validate() error {
+	if cluster.Status != nil && !validateVirtualizationStatus(*cluster.Status) {
+		return ErrInvalidVirtualizationStatus
+	}
+	return nil
+}
+
+// VirtualizationVirtualMachine represents a Virtualization Virtual Machine
+type VirtualizationVirtualMachine struct {
+	ID          int                    `json:"id,omitempty"`
+	Name        string                 `json:"name,omitempty"`
+	Status      *string                `json:"status,omitempty"`
+	Site        *DcimSite              `json:"site,omitempty"`
+	Cluster     *VirtualizationCluster `json:"cluster,omitempty"`
+	Role        *DcimDeviceRole        `json:"role,omitempty" mapstructure:"role"`
+	Device      *DcimDevice            `json:"device,omitempty"`
+	Platform    *DcimPlatform          `json:"platform,omitempty"`
+	PrimaryIPv4 *IpamIPAddress         `json:"primary_ip4,omitempty" mapstructure:"primary_ip4"`
+	PrimaryIPv6 *IpamIPAddress         `json:"primary_ip6,omitempty" mapstructure:"primary_ip6"`
+	Vcpus       *int                   `json:"vcpus,omitempty"`
+	Memory      *int                   `json:"memory,omitempty"`
+	Disk        *int                   `json:"disk,omitempty"`
+	Description *string                `json:"description,omitempty"`
+	Comments    *string                `json:"comments,omitempty"`
+	Tags        []*Tag                 `json:"tags,omitempty"`
+}
+
+// VirtualizationVMInterface represents a Virtualization Interface
+type VirtualizationVMInterface struct {
+	ID             int                           `json:"id,omitempty"`
+	VirtualMachine *VirtualizationVirtualMachine `json:"virtual_machine,omitempty"`
```

Comment:
Needs following change, otherwise virtual machine won't be unserialized properly
```suggestion
	VirtualMachine *VirtualizationVirtualMachine `json:"virtual_machine,omitempty"  mapstructure:"virtual_machine"`
```